### PR TITLE
Create github action publish.yml

### PR DESCRIPTION
**Pull request by diversemix**
Add a publish action

ref: https://github.com/libero/reviewer/issues/547

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Publish
+
+# Controls when the action will run. Triggers the workflow on push or pull request 
+# events but only for the master branch
+on:
```
Comment (giorgiosironi) [inline .github/workflows/publish.yml:3]: would stick to the painfully put together https://github.com/libero/reviewer-submission/blob/master/.github/workflows/ci.yml#L2-L7

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```
Comment (giorgiosironi) [inline .github/workflows/publish.yml]: move `if` into the last step, so that `make build` is executed on PRs

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Build Docker image
+        run: make build
+      - name: Push image
+        run: .scripts/github/push-image.sh reviewer-reporter
```
Comment (giorgiosironi) [inline .github/workflows/publish.yml:22]: this script assumes a `IMAGE_TAG` variable named like the commit, see https://github.com/libero/reviewer-submission/blob/master/.github/workflows/ci.yml#L8-L9
`make build` (or what is called inside that target) should read `IMAGE_TAG` too to tag the image appropriately

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (giorgiosironi) [inline .github/workflows/publish.yml:25]: is this where the Docker Hub token ended up?

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Build Docker image
+        run: make build
```
Comment (giorgiosironi) [inline .github/workflows/publish.yml:19]: https://github.com/libero/reviewer-reporter/blob/master/docker-compose.yaml#L7 should be `libero/...`; the `liberoadmin/...` rename is done by the push script and was always supposed to be temporary

File: .github/workflows/publish.yml (hunk 1 above)
Comment (diversemix) [reply .github/workflows/publish.yml:3]: Ah thanks :+1:

File: .github/workflows/publish.yml (hunk 3 above)
Comment (diversemix) [reply .github/workflows/publish.yml:22]: Ah ok- will change the docker-compose file appropriately too

File: .github/workflows/publish.yml (hunk 5 above)
Comment (diversemix) [reply .github/workflows/publish.yml:19]: Changed :+1:

Comment (diversemix) [reply .github/workflows/publish.yml:19]: Changed and raised https://github.com/libero/reviewer/issues/552

**Comment by diversemix** [comment]
@giorgiosironi / @cory 

Have we encountered this before? Looks like it errors with 

```
 ##[error]The input 'submodules' is not supported in actions/checkout@v2
```

This appears to be because we are attempting to use ssh for submodule on a private repo... there is a soln here: https://github.com/actions/checkout/issues/14#issuecomment-553383594

**Comment by giorgiosironi** [comment]
Stick to HTTPS rather than SSH for submodules:
https://github.com/libero/reviewer-submission/blob/master/.gitmodules
Travis CI had similar problems, as there was no private key configured, and we kept it working the same way.

**Review by Cooryd — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+name: Publish
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master events but only for the master branch
```
Comment (giorgiosironi) [inline .github/workflows/publish.yml]: is that a branch name?

Comment (diversemix) [reply .github/workflows/publish.yml]: No its not! Good catch :+1: 

Done!

**Review by giorgiosironi — APPROVED** [review]

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**